Orange/widgets/evaluate/owconfusionmatrix.py

```diff
@@ -89,9 +89,12 @@ class OWConfusionMatrix(widget.OWWidget):
     icon = "icons/ConfusionMatrix.svg"
     priority = 1001
 
-    inputs = [("Evaluation Results", Orange.evaluation.Results, "set_results")]
-    outputs = [("Selected Data", Orange.data.Table, widget.Default),
-               (ANNOTATED_DATA_SIGNAL_NAME, Orange.data.Table)]
+    class Inputs:
+        evaluation_results = Input("Evaluation Results", Orange.evaluation.Results)
+
+    class Outputs:
+        selected_data = Output("Selected Data", Orange.data.Table, default=True)
+        adsn = Output(ANNOTATED_DATA_SIGNAL_NAME, Orange.data.Table)
```

Comment:
And perhaps in general don't use non-standard abbreviations, acronyms in particular!